homeassistant/components/sensor/nederlandse_spoorwegen.py

```diff
@@ -0,0 +1,136 @@
+"""
+Support for Nederlandse Spoorwegen public transport.
+
+For more details on this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.nederlandse_spoorwegen/
+"""
+from datetime import datetime
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, ATTR_ATTRIBUTION
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['nsapi==2.7.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ATTRIBUTION = "Data provided by NS"
+CONF_DEPARTURES = 'departures'
+CONF_FROM = 'from'
+CONF_TO = 'to'
+CONF_VIA = 'via'
+CONF_EMAIL = 'email'
+CONF_PASSWORD = 'password'
+
+ICON = 'mdi:train'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=120)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_EMAIL): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_DEPARTURES): [{
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_FROM): cv.string,
+        vol.Required(CONF_TO): cv.string,
+        vol.Optional(CONF_VIA): cv.string}]
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the departure sensor."""
+    import ns_api
+    nsapi = ns_api.NSAPI(
+        config.get(CONF_EMAIL), config.get(CONF_PASSWORD))
```

Comment:
I assume that there will be an exception if the credentials are wrong. You need to catch that and make the platform setup fails.

Reply:
At this stage, it's merely storing the credentials. But I see your point that we should validate that before we add sensors (devices) that keep failing. I'm patching the code now to retrieve all valid station names at setup_platform level to see if the credentials are fine. Bonus points: this way I can also validate the station input by users.

Reply:
Sounds like a smart way to do it.